pom.xml

```diff
@@ -195,6 +200,14 @@
         <plugin>
           <artifactId>maven-surefire-plugin</artifactId>
           <version>2.18.1</version>
+          <dependencies>
+            <dependency>
+              <!-- Force junit provider because surefire will choose the testng one if testng is detected -->
+              <groupId>org.apache.maven.surefire</groupId>
+              <artifactId>surefire-junit47</artifactId>
```

Comment:
https://github.com/easymock/easymock/blob/master/core/src/test/java/org/easymock/tests2/EasyMockAnnotationsTest.java

It contains some JUnit tests and is detected as a JUnit class as "expected" by `TestNGDirectoryTestSuite#isJUnitTest`. But it is a bit incoherent with `junit=false` and a flag may be raised somewhere because `junit=false` + `junit=true` is weird.

TestNG has a confirmed issue with abstract class https://github.com/cbeust/testng/issues/1222 but I think surefire could do something too and not send such kind of classes :)